### PR TITLE
chore(ci): [NMP-27] Feature** Added eslint(airbnb) and prettier to project

### PR DESCRIPTION
**Pull request by kcaparas1630**
# Description

Added Eslint and prettier to have an easier time identifying and rectifying potential errors 

## Type of changes

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-43-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-43-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Review by DerekRoberts — APPROVED** [review]
Very nice changes.  :)

**Comment by kcaparas1630** [comment]
@LocalNewsTV I fixed configuration for eslintrc to follow airbnb style guide.

**Review by LocalNewsTV — COMMENTED** [review]
In pulling this on my own device the `npm run lint` didn't run until I installed two more dependencies not in your `package.json` outlined by the script 
`npm install eslint-plugin-react@latest --save-dev`
`npm install eslint-plugin-jsx-a11y@latest --save-dev`

This will also flag a bunch of issues since the default vite template doesn't follow AirBnB styling. but because its small and you started early, this is minimal refactoring. You'll be deleting 95% of it anyways because its just boilerplate.

Ultimately you're probably going to want to disable some of the rules, it's not a one size fits all, and you're using TS over JS 

This file is a year old, but here is some of the overrides we used when we were doing Capstone [You can see them here](https://github.com/bcgov/citz-imb-wayfinder/blob/main/src/frontend-pwa/.eslintrc.json)

**Comment by kcaparas1630** [comment]
I did some researching and referencing of what you did @LocalNewsTV from your past capstone, and added a couple of comments to provide understanding on what each line is doing. I've run npm run lint and it is giving me my lint problems. We'll be adding more once we get into creating the project further. Meanwhile, I'll settle for this initial settings for mostly TypeScript configurations.

Once approved I'll resolve conflicts with package dependencies. I'm wondering if resolving it needs to be on vscode, and merge it there to resolve conflicts on package.json and do a npm install to lock the dependencies.

**Comment by LocalNewsTV** [comment]
@kcaparas1630 
If you restart VSC after installing the dependencies that were missing it will pick them up on load. I've tested that it runs on my local
I don't know why they don't get picked up from the initial installation, It might just be how the extensions are configured, this isn't unique to your repo. 

There is also `Prettier` and `ESLint` extensions for VSC :)

You can also add recommended VSC Extensions to your repo, if you want to alert people developing to install them [Here's more details](https://dev.to/askrishnapravin/recommend-vs-code-extensions-to-your-future-teammates-4gkb).
I'm not asking you to do this in your PR, but it's always nice to know what options you have, if you don't know about them!

**Review by LocalNewsTV — APPROVED** [review]
I can verify locally that the ESLint and Prettier are functioning as expected once packages have been installed and VSC restarted.

Several linting issues, but majority of them are in the generated files that are going to be deleted in the next PR. As Continuous Deployment isn't actively setup. this is fine at this stage. Recommend prioritizing removing the unnecessary generated script, and making remainder of files comply to your linting rules.

The AirBNB style guide isn't a flawless design, you'll likely find limitations in the coming days, or that some rules don't work for what you need, expect to edit some of the rules over your next PR's. one that stands out is the `do not import modules using an absolute path` because your `public` folder uses them :) 

Either way, good job getting this set up and running.

File: frontend/.prettierrc.json [hunk 1]
```diff
@@ -1 +1,9 @@
-{}
+{
+    "semi": false,
+    "singleQuote": true,
+    "tabWidth": 2,
+    "trailingComma": "all",
+    "arrowParens": "always",
+    "printWidth": 100
+  }
+
```
Comment (LocalNewsTV) [inline frontend/.prettierrc.json:9]: You have an indent on this line, triggering the Error :)

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	// List of extensions which should be recommended for users of this workspace.
+	"recommendations": [
+		"dbaeumer.vscode-eslint",
+  		"esbenp.prettier-vscode"
+	],
+	// List of extensions recommended by VS Code that should not be recommended for users of this workspace.
+	"unwantedRecommendations": [
+		
+	]
+}
```
Comment (LocalNewsTV) [inline .vscode/extensions.json:14]: No blank end of line file. 
A lot of generated scripts don't put them in

**Pull request merged**